### PR TITLE
Feature/#1194

### PR DESCRIPTION
**Pull request by Mi-Lan**
issue #1194

### CODE REVIEW

**Comment by pimato** [comment]
@Mi-Lan you should be able to show the data now at least on Rinkeby: https://thegraph.com/explorer/subgraph/protofire/omen-rinkeby

**Comment by pimato** [comment]
the time should always be `HH - MM`, sometimes it shows:
Bildschirmfoto 2020-11-11 um 17 54 33

**Comment by pimato** [comment]
Bildschirmfoto 2020-11-11 um 17 56 28

should be a hyperlink pointing to etherscan on rinkeby or Mainnet!

**Comment by pimato** [comment]
are the shares/pool tokens shown correctly shown? 

Bildschirmfoto 2020-11-11 um 17 58 38

**Comment by pimato** [comment]
this:
Bildschirmfoto 2020-11-11 um 18 00 15

is not according my design specification:
Bildschirmfoto 2020-11-11 um 18 00 37

**Comment by pimato** [comment]
The buttons are not according my design specification:
Bildschirmfoto 2020-11-11 um 18 01 48

**Comment by pimato** [comment]
The navigation buttons are outdated:
Bildschirmfoto 2020-11-11 um 18 03 31

**Comment by Mi-Lan** [comment]
> are the shares/pool tokens shown correctly shown?
> 
> also this market here shows DAI in amount but it should be cDAI:
> Bildschirmfoto 2020-11-11 um 17 58 38
> 
> http://localhost:3000/#/0x8e550d69a3b5d3481fed7323d7941b5b5a2a639d

maybe we should put USD instead? Because that piece of data is collateralAmountUSD and i just get token address so i would have to map it for DAI,Cdai,USDT...

**Comment by pimato** [comment]
> > are the shares/pool tokens shown correctly shown?
> > also this market here shows DAI in amount but it should be cDAI:
> > Bildschirmfoto 2020-11-11 um 17 58 38
> > http://localhost:3000/#/0x8e550d69a3b5d3481fed7323d7941b5b5a2a639d
> 
> @pimato maybe we should put USD instead? Because that piece of data is collateralAmountUSD and i just get token address so i would have to map it for DAI,Cdai,USDT...Also i didn't know how to format that collateralAmount i saw in other places how it was done and copied maybe i'm doing it wrong..

USD does not make sense. why not use the `collateralTokenAmount`?

**Comment by pimato** [comment]
> The buttons are not according my design specification:
> Bildschirmfoto 2020-11-11 um 18 01 48
> 
> should:
> Bildschirmfoto 2020-11-11 um 18 02 26

this has not been fixed.

**Comment by pimato** [comment]
> this:
> Bildschirmfoto 2020-11-11 um 18 00 15
> 
> is not according my design specification:
> Bildschirmfoto 2020-11-11 um 18 00 37

this has not been fixed.

File: app/src/common/constants.ts
Comment (Mi-Lan) [inline app/src/common/constants.ts:57]: this needs to be replaced once it is ready for mege!

**Comment by KadenZipfel** [comment]
As mentioned in the corresponding subgraph PR: https://github.com/protofire/omen-subgraph/pull/93#issuecomment-733879195, the subgraph change used to track transaction details used here is incomplete. Realistically, this still works pretty well with this PR, though it does hide some data that could be important to the user.

**Comment by Mi-Lan** [comment]
> As mentioned in the corresponding subgraph PR: [protofire/omen-subgraph#93 (comment)](https://github.com/protofire/omen-subgraph/pull/93#issuecomment-733879195), the subgraph change used to track transaction details used here is incomplete. Realistically, this still works pretty well with this PR, though it does hide some data that could be important to the user.

@KadenZipfel @pimato i don't know how we could represent that data then? It might be confusing to user to see 2 changes with one action? What do you guys think?

**Comment by pimato** [comment]
> > As mentioned in the corresponding subgraph PR: [protofire/omen-subgraph#93 (comment)](https://github.com/protofire/omen-subgraph/pull/93#issuecomment-733879195), the subgraph change used to track transaction details used here is incomplete. Realistically, this still works pretty well with this PR, though it does hide some data that could be important to the user.
> 
> @KadenZipfel @pimato i don't know how we could represent that data then? It might be confusing to user to see 2 changes with one action? What do you guys think? Or we could just change how liquidity transaction is displayed on table so that it encompasses that one extra buy action?

I think it should be a two rows, one for deposit and one for buy.

**Comment by Mi-Lan** [comment]
> > > As mentioned in the corresponding subgraph PR: [protofire/omen-subgraph#93 (comment)](https://github.com/protofire/omen-subgraph/pull/93#issuecomment-733879195), the subgraph change used to track transaction details used here is incomplete. Realistically, this still works pretty well with this PR, though it does hide some data that could be important to the user.
> > 
> > 
> > @KadenZipfel @pimato i don't know how we could represent that data then? It might be confusing to user to see 2 changes with one action? What do you guys think? Or we could just change how liquidity transaction is displayed on table so that it encompasses that one extra buy action?
> 
> I think it should be a two rows, one for deposit and one for buy.

@pimato and what will be with rows for buy and sell which don't have those amounts? Should they just be empty?

**Comment by KadenZipfel** [comment]
Yeah, I think we should really show all the actions that are happening. It may be confusing at first, but at least that way users can see what's actually going and know that they own outcome shares, even though they may not have purchased them themselves. I think the proper solution here is to properly set this up on the subgraph if possible.

**Comment by pimato** [comment]
@Mi-Lan I am getting an error on market creation which might be related with the new rinkeby proxy contracts:
Bildschirmfoto 2020-12-09 um 17 46 35

**Review by pimato — CHANGES_REQUESTED** [review]
see comments

**Comment by Mi-Lan** [comment]
> @Mi-Lan I am getting an error on market creation which might be related with the new rinkeby proxy contracts:
> Bildschirmfoto 2020-12-09 um 17 46 35

@pimato @KadenZipfel yeah i think this has nothing to do with my pr i haven't made any changes on that part...when we merge subgraph pr i think this will get fixed

**Review by hexyls — CHANGES_REQUESTED** [review]
This is so cool!

Hmmm so the error mentioned above appears to only happen for me when loading scalar markets.. This PR is using the astro514/omen subgraph for now which is prob missing some scalar stuff? Not sure.

Anyway, I am testing this PR by loading [this rinkeby market](http://localhost:3000/#/0xcff1ddf53f7f9ca5948e2c842532e5805878891a)

It looks great! I did notice that a highlight is missing on the Activities button when it is selected, for example:

Screen Shot 2020-12-11 at 3 18 36 pm

Should be:

Screen Shot 2020-12-11 at 3 19 26 pm

One general question I have for everyone that has more to do with the subgraph than this PR - do we want the user address listed for each activity to be the address of the user or the address of the user's proxy? In the screenshot below my proxy is listed as the last buyer instead of my actual address.


Screen Shot 2020-12-11 at 3 24 15 pm

**Comment by pimato** [comment]
> This is so cool!
> 
> Hmmm so the error mentioned above appears to only happen for me when loading scalar markets.. This PR is using the astro514/omen subgraph for now which is prob missing some scalar stuff? Not sure.
> 
> Anyway, I am testing this PR by loading [this rinkeby market](http://localhost:3000/#/0xcff1ddf53f7f9ca5948e2c842532e5805878891a)
> 
> It looks great! I did notice that a highlight is missing on the Activities button when it is selected, for example:
> 
> Screen Shot 2020-12-11 at 3 18 36 pm
> 
> Should be:
> 
> Screen Shot 2020-12-11 at 3 19 26 pm
> 
> One general question I have for everyone that has more to do with the subgraph than this PR - do we want the user address listed for each activity to be the address of the user or the address of the user's proxy? In the screenshot below my proxy is listed as the last buyer instead of my actual address.
> 
> Screen Shot 2020-12-11 at 3 24 15 pm

Great points @hexyls !

regarding the address, the proxy is just a tool/assistant which is in the background and should stay in the background. With that, I would recommend to change the address to the proxy owner.

**Review by pimato — CHANGES_REQUESTED** [review]
see comments

**Review by hexyls — APPROVED** [review]
Looks good once conflicts are resolved! 🏆

File: app/src/util/tools.ts
Comment (KadenZipfel) [inline app/src/util/tools.ts:386]: All `tools` functions should have corresponding tests in `tools.spec.ts`

**Review by KadenZipfel — CHANGES_REQUESTED** [review]
Overall this is really good! 

I did try a deposit transaction and it hasn't been included in the activities list for some reason, even after waiting a few minutes and refreshing. Maybe a subgraph issue?

File: app/src/services/market_maker.ts
Comment (Mi-Lan) [reply app/src/services/market_maker.ts:62]: You are right! Renaming it

File: app/src/components/market/common/history_section/historySelect.tsx
Comment (Mi-Lan) [reply app/src/components/market/common/history_section/historySelect.tsx:107]: Hmm yeah that would be good but it would require extra evaluation step before formatting since as you can see it has two return statments.
Screenshot 2021-01-12 at 17 08 25

File: app/src/components/market/common/history_section/historySelect.tsx
Comment (Mi-Lan) [reply app/src/components/market/common/history_section/historySelect.tsx:141]: Ohh yeah good call!

File: app/src/components/market/common/history_section/index.tsx
Comment (Mi-Lan) [reply app/src/components/market/common/history_section/index.tsx:20]: Ohh yeah subgraph can be queried by the blockNumber! Though i didn't create that...

File: app/src/util/tools.ts
Comment (Mi-Lan) [reply app/src/util/tools.ts:386]: aha will look into it though i don't have experience with creating tests

File: app/src/hooks/useGraphFpmmTransactionsFromQuestion.tsx
Comment (Mi-Lan) [reply app/src/hooks/useGraphFpmmTransactionsFromQuestion.tsx:112]: I fixed this although i didn't BigNumberfy them since when i include BigNumber types with typescript it does that automatically. Is there other reason to do it?

File: app/src/components/market/common/history_chart/index.tsx
Comment (pRoy24) [inline app/src/components/market/common/history_chart/index.tsx:139]: Can we set these colors in themes and import from there?
I imagine there might be multiple charts in the future that would want to have the same look and feel

File: app/src/components/market/common/history_section/index.tsx
Comment (pRoy24) [inline app/src/components/market/common/history_section/index.tsx:141]: This line throws an error when switching to "All" in the filter picker and the app crashes, because the totalDataPoints is negative so the range function throws an error.
I am guessing that it has to do with the datapoints returned by "All" which is negative if the market hasn't existed for a long time.

Steps to reproduce
Switch to Rinkeby tab
Go to market /#/0xf9cefa4ade74834882589f5309664c35ecdeff5f
History->Graph.
Click a few date range filters and finally click the "All" filter.
image

Comment (Mi-Lan) [reply app/src/components/market/common/history_section/index.tsx:141]: I made issue for that that is on master as well. Here is the issue https://github.com/protofire/omen-exchange/issues/1536

File: app/src/components/market/common/history_table/index.tsx
Comment (pRoy24) [inline app/src/components/market/common/history_table/index.tsx:122]: Perhaps best to use externally defined constants for these values.

File: app/src/util/tools.ts
Comment (pRoy24) [inline app/src/util/tools.ts:363]: Why is the month value 0 for me?
image
Maybe the month value needs to be offset by 1?

Comment (pRoy24) [inline app/src/util/tools.ts:363]: Also shouldn't the date string be formatted as dd/mm instead of dd.mm ? Its not obvious to me that value represents a date.

**Review by pRoy24 — CHANGES_REQUESTED** [review]
Hey man, I am a bit late to the party so don't have too much context and background but got into an error when testing locally on Rinkeby (see comment) Also in the activity table the month shows as 0 -11 range for me instead of 1/12 range eg. Jan shows as 0.

Let me know if I am missing something.

File: app/src/components/market/common/history_section/index.tsx
Comment (pRoy24) [reply app/src/components/market/common/history_section/index.tsx:141]: So I guess you'll either need to handle it on your side or wait for a fix to be made on the other side before deploying.

Comment (Mi-Lan) [reply app/src/components/market/common/history_section/index.tsx:141]: Yeah but that is not the error caused by this pr and this pr has been expanded enough already.And it won't make any difference since it already has the same error on master...

File: app/src/components/market/common/history_table/index.tsx
Comment (Mi-Lan) [reply app/src/components/market/common/history_table/index.tsx:122]: Hmmm i think we don't have that do you have some suggestion on how to implement it?

File: app/src/util/tools.ts
Comment (Mi-Lan) [reply app/src/util/tools.ts:363]: Ohh yeah good catch haven't noticed that will fix it!

Comment (Mi-Lan) [reply app/src/util/tools.ts:363]: You mean with the dash? Well i was basing it on figma designs here is how it is there...
Screenshot 2021-01-13 at 15 23 41 
- Here is the figma link it should be on right side https://www.figma.com/file/R056tSYMjvfwaFidEBItYy/Widgets-V2?node-id=1209%3A141

File: app/src/components/market/common/history_table/index.tsx
Comment (Mi-Lan) [reply app/src/components/market/common/history_table/index.tsx:100]: Should i then refactor the whole table since i made it with divs?

File: app/src/components/market/common/history_chart/index.tsx
Comment (Mi-Lan) [reply app/src/components/market/common/history_chart/index.tsx:139]: well i would need to import theme in component and then use it like that since i don't have access to themes in this scope should i do that?

Comment (Mi-Lan) [reply app/src/components/market/common/history_chart/index.tsx:139]: Screenshot 2021-01-13 at 23 12 14
Is this good? I haven't seen that we use themes like this but does work....

File: app/src/components/market/common/history_table/index.tsx
Comment (pRoy24) [reply app/src/components/market/common/history_table/index.tsx:121]: Not a big deal perhaps you could define an enum in types.ts

export enum EterscanBase = {
  rinkebyBase = 'https://rinkeby.etherscan.io/tx/',
  mainnetBase = 'https://etherscan.io/tx/'
}

**Review by pRoy24 — APPROVED** [review]
LGTM once the Filter error for "All" type is resolved.

File: app/src/components/market/common/history_table/index.tsx
Comment (Mi-Lan) [reply app/src/components/market/common/history_table/index.tsx:121]: Ohhh you meant that. I misunderstood your comment. I thought you meant like having globally accessible place for figuring out which network we are on currently... for some reason :D Will include it!

**Comment by Mi-Lan** [comment]
In this laterst commit i included fix for chart error #1536 which was cause by using earliest mainnet block on rinkeby

File: app/src/components/market/common/history_section/history_select.tsx
Comment (KadenZipfel) [inline app/src/components/market/common/history_section/history_select.tsx:200]: Shouldn't we already have access to these values?

File: app/src/hooks/useGraphFpmmTransactionsFromQuestion.tsx
Comment (KadenZipfel) [reply app/src/hooks/useGraphFpmmTransactionsFromQuestion.tsx]: When they're pulled from the subgraph I think they should be strings, strange... Do they work normally as `BigNumber`s?

**Review by KadenZipfel — COMMENTED** [review]
.

**Comment by KadenZipfel** [comment]
Found a bug: 
- Withdraw liquidity from an unequal pool, i.e. not 50/50 odds
- Outcome shares are being purchased behind the scenes as a result of the liquidity withdrawal
- History displays my liquidity withdrawal along with a 'sell' which should be a 'buy'. I also noticed that the amount of outcome shares displayed in the table is also incorrect.

I haven't dug into the code for this, but looking at other liquidity withdrawal transactions, it seems like it was probably assumed that additional outcome shares are being sold, but in fact every liquidity transaction in a non-equal pool results in a purchase of one or the other outcome shares. Maybe the reason the amount is wrong is because you're flipping buy/sell between adding/removing liquidity when you should just be flipping which outcome is being purchased.

File: app/src/components/market/common/history_section/history_select.tsx
Comment (Mi-Lan) [reply app/src/components/market/common/history_section/history_select.tsx:200]: we don't have them in there actually....

File: app/src/hooks/useGraphFpmmTransactionsFromQuestion.tsx
Comment (Mi-Lan) [reply app/src/hooks/useGraphFpmmTransactionsFromQuestion.tsx]: yeah they work great i even have to convert them when using so all is good!

**Comment by Mi-Lan** [comment]
> Found a bug:
> 
> * Withdraw liquidity from an unequal pool, i.e. not 50/50 odds
> * Outcome shares are being purchased behind the scenes as a result of the liquidity withdrawal
> * History displays my liquidity withdrawal along with a 'sell' which should be a 'buy'. I also noticed that the amount of outcome shares displayed in the table is also incorrect.
> 
> I haven't dug into the code for this, but looking at other liquidity withdrawal transactions, it seems like it was probably assumed that additional outcome shares are being sold, but in fact every liquidity transaction in a non-equal pool results in a purchase of one or the other outcome shares. Maybe the reason the amount is wrong is because you're flipping buy/sell between adding/removing liquidity when you should just be flipping which outcome is being purchased.

hmm yeah i was also testing it now seems that withdrawal doesn't come out good...will look more into it

**Comment by Mi-Lan** [comment]
> Found a bug:
> 
> * Withdraw liquidity from an unequal pool, i.e. not 50/50 odds
> * Outcome shares are being purchased behind the scenes as a result of the liquidity withdrawal
> * History displays my liquidity withdrawal along with a 'sell' which should be a 'buy'. I also noticed that the amount of outcome shares displayed in the table is also incorrect.
> 
> I haven't dug into the code for this, but looking at other liquidity withdrawal transactions, it seems like it was probably assumed that additional outcome shares are being sold, but in fact every liquidity transaction in a non-equal pool results in a purchase of one or the other outcome shares. Maybe the reason the amount is wrong is because you're flipping buy/sell between adding/removing liquidity when you should just be flipping which outcome is being purchased.

@KadenZipfel I was looking at this and researching it today seems that the issue is that some liquidity transactions collateralAmount is zero which messes my formula up...

**Comment by KadenZipfel** [comment]
> > Found a bug:
> > 
> > * Withdraw liquidity from an unequal pool, i.e. not 50/50 odds
> > * Outcome shares are being purchased behind the scenes as a result of the liquidity withdrawal
> > * History displays my liquidity withdrawal along with a 'sell' which should be a 'buy'. I also noticed that the amount of outcome shares displayed in the table is also incorrect.
> > 
> > I haven't dug into the code for this, but looking at other liquidity withdrawal transactions, it seems like it was probably assumed that additional outcome shares are being sold, but in fact every liquidity transaction in a non-equal pool results in a purchase of one or the other outcome shares. Maybe the reason the amount is wrong is because you're flipping buy/sell between adding/removing liquidity when you should just be flipping which outcome is being purchased.
> 
> @KadenZipfel I was looking at this and researching it today seems that the issue is that some liquidity transactions collateralAmount is zero which messes my formula up but i don't display those...And the sell was just hardcoded wrongly by me. Can you maybe give example where it's calculated wrongly?
> Screenshot 2021-01-15 at 17 45 57
> as you can see from screenshot field collateralTokenAmount which we get from subgraph....

Yeah, looking at the subgraph it should be returning a value for `collateralTokenAmount`... Can you dig into that a little bit and make an issue in the subgraph repo?

**Comment by Mi-Lan** [comment]
> > > Found a bug:
> > > 
> > > * Withdraw liquidity from an unequal pool, i.e. not 50/50 odds
> > > * Outcome shares are being purchased behind the scenes as a result of the liquidity withdrawal
> > > * History displays my liquidity withdrawal along with a 'sell' which should be a 'buy'. I also noticed that the amount of outcome shares displayed in the table is also incorrect.
> > > 
> > > I haven't dug into the code for this, but looking at other liquidity withdrawal transactions, it seems like it was probably assumed that additional outcome shares are being sold, but in fact every liquidity transaction in a non-equal pool results in a purchase of one or the other outcome shares. Maybe the reason the amount is wrong is because you're flipping buy/sell between adding/removing liquidity when you should just be flipping which outcome is being purchased.
> > 
> > 
> > @KadenZipfel I was looking at this and researching it today seems that the issue is that some liquidity transactions collateralAmount is zero which messes my formula up but i don't display those...And the sell was just hardcoded wrongly by me. Can you maybe give example where it's calculated wrongly?
> > Screenshot 2021-01-15 at 17 45 57
> > as you can see from screenshot field collateralTokenAmount which we get from subgraph....
> 
> Yeah, looking at the subgraph it should be returning a value for `collateralTokenAmount`... Can you dig into that a little bit and make an issue in the subgraph repo?

@KadenZipfel Yeah i will make issue for it!

File: app/src/services/market_maker.ts
Comment (Mi-Lan) [inline app/src/services/market_maker.ts:169]: @KadenZipfel hey men this is the problematic function that doesn't work on xDai.

File: app/src/services/conditional_token.ts
Comment (KadenZipfel) [inline app/src/services/conditional_token.ts:79]: If you set `positionId: BigNumber` it works :)
Not sure if it causes any other problems but I don't think it should

Comment (Mi-Lan) [reply app/src/services/conditional_token.ts]: Nice find!

**Comment by KadenZipfel** [comment]
Withdraw transactions seem to be missing the collateral amount
Screen Shot 2021-02-09 at 10 50 35 AM

**Comment by Mi-Lan** [comment]
> Withdraw transactions seem to be missing the collateral amount
> Screen Shot 2021-02-09 at 10 50 35 AM

On which market is that? I'm having trouble loading the dai markets....In cases where i have seen that its zero its getting it from subgraph....have you checked maybe?

**Comment by KadenZipfel** [comment]
> > Withdraw transactions seem to be missing the collateral amount
> > Screen Shot 2021-02-09 at 10 50 35 AM
> 
> @KadenZipfel On which market is that? I'm having trouble loading the dai markets....In cases where i have seen that its zero its getting it from subgraph....have you checked maybe?

I'm seeing it on these Rinkeby markets: 
http://localhost:3000/#/0x03fd91b7385c0a2fcbbab86bd7342861ba0c6028
http://localhost:3000/#/0xae104263996afa57e715d2a70a9641a9947cc4ea

**Comment by Mi-Lan** [comment]
> > > Withdraw transactions seem to be missing the collateral amount
> > > Screen Shot 2021-02-09 at 10 50 35 AM
> > 
> > 
> > @KadenZipfel On which market is that? I'm having trouble loading the dai markets....In cases where i have seen that its zero its getting it from subgraph....have you checked maybe?
> 
> I'm seeing it on these Rinkeby markets:
> http://localhost:3000/#/0x03fd91b7385c0a2fcbbab86bd7342861ba0c6028
> http://localhost:3000/#/0xae104263996afa57e715d2a70a9641a9947cc4ea

Yeah i looked into it seems thats the same issue we had earlier that collateralAmount is very small
Screenshot 2021-02-10 at 23 51 32
- On this screenshot on far right you can see that number is much smaller then the colalteralAmount for the first item...
- @KadenZipfel

**Comment by KadenZipfel** [comment]
> > > > Withdraw transactions seem to be missing the collateral amount
> > > > Screen Shot 2021-02-09 at 10 50 35 AM
> > > 
> > > 
> > > @KadenZipfel On which market is that? I'm having trouble loading the dai markets....In cases where i have seen that its zero its getting it from subgraph....have you checked maybe?
> > 
> > 
> > I'm seeing it on these Rinkeby markets:
> > http://localhost:3000/#/0x03fd91b7385c0a2fcbbab86bd7342861ba0c6028
> > http://localhost:3000/#/0xae104263996afa57e715d2a70a9641a9947cc4ea
> 
> Yeah i looked into it seems thats the same issue we had earlier that collateralAmount is very small
> Screenshot 2021-02-10 at 23 51 32
> 
> * On this screenshot on far right you can see that number is much smaller then the colalteralAmount for the first item...
> * @KadenZipfel

Yeah, seems like a subgraph issue. I made an issue https://github.com/protofire/omen-subgraph/issues/145 and will look into it today if I have time.

**Comment by KadenZipfel** [comment]
Should be fixed by: https://github.com/protofire/omen-subgraph/pull/146
You can test on `kadenzipfel/omen-rinkeby` in the meantime

**Comment by Mi-Lan** [comment]
@pimato @KadenZipfel Seems that that subgraph pr will fix the zero value! But there is case on scalar markets when you deposit small amount(0.01 DAI) of additional shares that its below 3 digits how should we handle that case? 
Screenshot 2021-02-16 at 14 25 49
- It says +0.00 but in reality its buying some small amount....
Screenshot 2021-02-16 at 14 26 33
- Heres the subgraph data for it 
Screenshot 2021-02-16 at 14 27 40
@KadenZipfel @pimato

**Comment by Mi-Lan** [comment]
I changed graph link for rinkeby to be Kadens so that it can be tested...This should be changes before merging

**Comment by KadenZipfel** [comment]
> @pimato @KadenZipfel Seems that that subgraph pr will fix the zero value! But there is case on scalar markets when you deposit small amount for example 0.01 DAI ...additional shares bought in that case are so small that they are not visible with 3 decimal places?
> Screenshot 2021-02-16 at 14 25 49
> 
> * It says +0.00 but in reality its buying some small amount....
>   Screenshot 2021-02-16 at 14 26 33
> * Heres the subgraph data for it
>   Screenshot 2021-02-16 at 14 27 40
>   @KadenZipfel @pimato

Yeah, that seems like it's returning a reasonable value. Maybe worth not displaying additional shares if it's only a very small amount

**Comment by Mi-Lan** [comment]
I will merge conflicts here once we are ready for merge with subgraph since there are changes made on the chart component which has different structure now with the changes...i'm assuming that the change is related to kadens calculations consolidation since i think one relates to function used in charts....

**Review by KadenZipfel — APPROVED** [review]
Great work on a super tough feature 👏

**Review by pimato — CHANGES_REQUESTED** [review]
Network: xDai
Looks like at some point the history transaction list stopped loading new markets:
Bildschirmaufnahme 2021-02-26 um 14 44 38

For this market: 
http://localhost:3000/#/0xd3f61725f4a1905223160abec29b98e81cd569ba

Error message:
Bildschirmfoto 2021-02-26 um 14 47 54

**Comment by pimato** [comment]
Disabled button is not according our design specification for disabled buttons:
Bildschirmfoto 2021-02-26 um 14 53 29

Should:
Bildschirmfoto 2021-02-26 um 14 53 38

**Comment by pimato** [comment]
Active buttons should not have onHover styling:
Bildschirmfoto 2021-02-26 um 14 55 08

**Comment by pimato** [comment]
`All` in the dropdown should be renamed to `All Activities` and the collapsed modal should have the same width as the dropdown button:
Bildschirmfoto 2021-02-26 um 14 55 43

Should:
Bildschirmfoto 2021-02-26 um 14 55 35

**Comment by pimato** [comment]
Every hyperlink should have a hover styling with color #5C6BC0
Bildschirmfoto 2021-02-26 um 14 59 55

**Review by pimato — CHANGES_REQUESTED** [review]
looks like I get an error while trying to load only `Trading` Activities in the History tab.

Market: http://localhost:3000/#/0x09ba10798a54f8830e3568a000e55981c493f043
Network: xDai

Bildschirmfoto 2021-03-01 um 10 59 06
Bildschirmfoto 2021-03-01 um 10 57 49

**Review by pimato — APPROVED** [review]

**Pull request merged**
